### ISSUE TITLE
MAP returning 'nan' in model.train

### ISSUE THREAD

**Issue by carlybatist** [kitzeslab/opensoundscape#522]
I am running the following code - 
```
labels = pd.read_csv('subset1.csv').set_index('filename') #already in one-hot-encoded-label format, with column names '0' & '1' for absence & presence of call, respectively
train_df, test_df = train_test_split(labels,test_size=0.2,random_state=2)
train_df1, validation_df = train_test_split(train_df, test_size = 0.2, random_state=2) 
classes = train_df1.columns

model = CNN('resnet50',classes=classes,sample_duration=10,single_target=True)

model.logging_level = 2 #3 request lots of logged content, 0 requests none
model.log_file = './binary_train2/training_log2.txt' #specify a file to log output to
Path(model.log_file).parent.mkdir(parents=True,exist_ok=True) #make the folder ./binary_train2

model.train(
    train_df=train_df1,
    validation_df=validation_df,
    save_path='./binary_train2/',
    epochs=5,
    batch_size=8,
    save_interval=1,
    num_workers=6,
)
```
And the MAP keeps getting calculated as 'nan' both for the training data & validation data 
<img width="126" alt="Screen Shot 2022-08-14 at 10 21 36 PM" src="https://user-images.githubusercontent.com/46716228/184580774-ed36b37a-fb3e-4569-beb7-4bdc3da91b13.png">
<img width="369" alt="Screen Shot 2022-08-14 at 9 16 58 PM" src="https://user-images.githubusercontent.com/46716228/184580871-c63259ab-9399-470c-a305-257be1ba4d39.png">

It is giving me these warning messages when the code is running but I don't know what they mean. 
<img width="948" alt="Screen Shot 2022-08-14 at 9 20 35 PM" src="https://user-images.githubusercontent.com/46716228/184580916-70594367-3060-468a-b7d1-645f268437e8.png">

**Comment by sammlapp** [comment]
The MAP metric will be undefined when there are no samples in a specific class for the validation (or training) set because of division by zero in the calculation of the metric. Therefore this is the expected behavior.
